src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -375,20 +379,37 @@ public void onRandomMapButtonClicked() {
     mapListView.scrollTo(mapIndex);
   }
 
-  public void onGenerateMapButtonClicked() {
+  public void onGenerateMapButtonClicked(){
+    onGenerateMap();
+  }
+
+  private void onGenerateMap() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
-        Platform.runLater(() -> {
-          initMapSelection();
-          mapListView.getItems().stream()
-              .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
-              .findAny().ifPresent(mapBean -> {
-            mapListView.getSelectionModel().select(mapBean);
-            mapListView.scrollTo(mapBean);
-            setSelectedMap(mapBean);
+      String generatorVersion = mapGeneratorService.queryMaxSupportedVersion();
+      String majorVersion = generatorVersion.split("\\.")[0];
+      if (majorVersion.equals("1")) {
+        GenerateMapController generateMapController = uiService.loadFxml("theme/play/generate_map.fxml");
+
+        Pane root = generateMapController.getRoot();
+        generateMapController.setCreateGameController(this);
+        JFXDialog dialog = uiService.showInDialog(gamesRoot, root, i18n.get("game.generate.dialog"));
```

Comment:
We are trying to get away from JFX since a few weeks. Can u use the notification and a Immediate notification with custom UI. @micheljung what should he do?

Reply:
Decided to keep JFXDialog for now.